features/collection/loaders/index.ts

```diff
@@ -30,13 +33,16 @@ export async function loadCollection(
       const ancestorsHandles = getAncestorHandles(hierarchy).reverse();
       ancestors = await loadCollections(ancestorsHandles, client);
    }
+
+   const news = await loadRelatedNews([collection.title]);
```

Comment:
Doing the await here means this whole component doesn't render until we've got the response from the API request. When doing async requests that result in data being displayed on the page, it's better to use a hook; something like [`useAsync`](https://usehooks.com/useAsync/).

Reply:
Especially since we're already in a nested request here. Using `useAsync` for both of these means we could render the component immediately (even if blank / empty) and not have to wait for two requests, one after the other.

Reply:
It’s also potentially worth using https://github.com/async-library/react-async as it makes managing the render state quite easy

Reply:
@danielbeardsley @jarstelfox I'm not sure what you mean here.. This request is made server side, when Next.js method `getServerSideProps` is called to get data for the page. You can't use hooks inside `getServerSideProps`.

Anyway If we want to skip SSR for related posts because we want the page to be served faster, we can lazy load them client side.

Reply:
@dhmacs Thanks for the clarification. I had missed this was part of SSR. I am less concerned about the performance issues here. I am not sure of this changes @danielbeardsley's mind.

Reply:
@danielbeardsley let me know if you have more questions, so that we can move this forward